core/trino-spi/src/test/java/io/trino/spi/block/BenchmarkCopyPositions.java

```diff
@@ -70,27 +72,36 @@ public Block copyPositions(BenchmarkData data)
         })
         private SelectedPositions selectedPositions;
 
+        @Param({"VARCHAR", "ROW(BIGINT)"})
+        private String type;
+
         private int[] positionsIds;
-        private Block variableWidthBlock;
+        private Block block;
 
-        public BenchmarkData(int selectedPositionsCount, boolean nullsAllowed, SelectedPositions selectedPositions)
+        public BenchmarkData(int selectedPositionsCount, boolean nullsAllowed, SelectedPositions selectedPositions, String type)
         {
             this.selectedPositionsCount = selectedPositionsCount;
             this.nullsAllowed = nullsAllowed;
             this.selectedPositions = selectedPositions;
+            this.type = type;
         }
 
         public BenchmarkData()
         {
-            this(1000, false, SelectedPositions.SEQUENCE);
+            this(1000, false, SelectedPositions.SEQUENCE, "VARCHAR");
         }
 
         @Setup
         public void setup()
         {
             positionsIds = selectedPositions.generateIds(selectedPositionsCount);
-            Slice[] slices = generateValues();
-            variableWidthBlock = createBlockBuilderWithValues(slices).build();
+            if (type.equals("VARCHAR")) {
+                Slice[] slices = generateValues();
+                block = createBlockBuilderWithValues(slices).build();
+            }
+            else if (type.equals("ROW(BIGINT)")) {
+                block = createRowBlock(POSITIONS, new LongArrayBlock(POSITIONS, Optional.empty(), LongStream.range(0, POSITIONS).toArray()));
```

Comment:
Although the contents of the LongArrayBlock likely don't matter for this change, still can we randomise the input `values` ?

Reply:
I added long randomization and the results are a little worse but still, there is an improvement.
I also added profile pollution to the benchmark to keep it more real.

```
Benchmark                             nullsAllowed  selectedPositions  selectedPositionsCount  type         baseline  branchless  branchless%
BenchmarkCopyPositions.copyPositions  FALSE         GROUPED            200                     ROW(BIGINT)  0.484     0.484       0
BenchmarkCopyPositions.copyPositions  FALSE         GROUPED            1000                    ROW(BIGINT)  2.082     2.083       0.04803073967
BenchmarkCopyPositions.copyPositions  FALSE         GROUPED            8000                    ROW(BIGINT)  18.698    18.893      1.042892288
BenchmarkCopyPositions.copyPositions  FALSE         SEQUENCE           200                     ROW(BIGINT)  0.484     0.492       1.652892562
BenchmarkCopyPositions.copyPositions  FALSE         SEQUENCE           1000                    ROW(BIGINT)  2.097     2.102       0.2384358608
BenchmarkCopyPositions.copyPositions  FALSE         SEQUENCE           8000                    ROW(BIGINT)  18.881    18.914      0.1747788782
BenchmarkCopyPositions.copyPositions  FALSE         RANDOM             200                     ROW(BIGINT)  0.489     0.492       0.6134969325
BenchmarkCopyPositions.copyPositions  FALSE         RANDOM             1000                    ROW(BIGINT)  2.181     2.145       -1.650618982
BenchmarkCopyPositions.copyPositions  FALSE         RANDOM             8000                    ROW(BIGINT)  21.12     20.017      -5.222537879
BenchmarkCopyPositions.copyPositions  TRUE          GROUPED            200                     ROW(BIGINT)  0.862     0.894       3.712296984
BenchmarkCopyPositions.copyPositions  TRUE          GROUPED            1000                    ROW(BIGINT)  3.968     4.005       0.9324596774
BenchmarkCopyPositions.copyPositions  TRUE          GROUPED            8000                    ROW(BIGINT)  41.909    27.061      -35.4291441
BenchmarkCopyPositions.copyPositions  TRUE          SEQUENCE           200                     ROW(BIGINT)  0.902     0.94        4.21286031
BenchmarkCopyPositions.copyPositions  TRUE          SEQUENCE           1000                    ROW(BIGINT)  4.389     4.174       -4.898610162
BenchmarkCopyPositions.copyPositions  TRUE          SEQUENCE           8000                    ROW(BIGINT)  41.143    28.406      -30.95787862
BenchmarkCopyPositions.copyPositions  TRUE          RANDOM             200                     ROW(BIGINT)  0.9       0.927       3
BenchmarkCopyPositions.copyPositions  TRUE          RANDOM             1000                    ROW(BIGINT)  4.556     4.17        -8.472344162
BenchmarkCopyPositions.copyPositions  TRUE          RANDOM             8000                    ROW(BIGINT)  41.507    29.262      -29.50104802
```